solana-programs/payment-router/programs/payment-router/src/lib.rs

```diff
@@ -70,23 +66,27 @@ pub struct CreatePDA<'info> {
     )]
     /// CHECK: This is the PDA owned by this program. This account will temporarily hold SOL USDC and SOL AUDIO tokens
     /// before transferring them over to given recipients, all within the same transaction.
-    pub payment_router_pda: AccountInfo<'info>,
+    pub payment_router_pda: UncheckedAccount<'info>,
```

Comment:
why was this necessary?

Reply:
I think this is due to my feedback on previous PRs - Anchor added "UncheckedAccount" as an alias for AccountInfo to make it very clear that there are no implicit checks on that account.

More context: https://solana.stackexchange.com/a/3780

I think it's fine in this case as that's what the example does for PDAs: [Programs as Signers](https://www.anchor-lang.com/docs/pdas)

> The authority account is now an UncheckedAccount instead of a Signer. When the puppet-master is invoked, the authority pda is not a signer yet so we mustn't add a check for it.

I am very wary of the amount of UncheckedAccount usage in both of these programs, but we have explicit checks so we should be good?

Reply:
Sounds like we should implement a custom type
https://solana.stackexchange.com/a/3819

Reply:
It just feels very dangerous, though it might not be explicitly right now. Future dev effort could use it blindly and not realize what's going on?